test_unstructured_ingest/python/test-ingest-astra-output.py

```diff
@@ -0,0 +1,57 @@
+import random
+
+import click
+from astrapy.db import AstraDB
+
+
+@click.command()
+@click.option("--token", type=str)
+@click.option("--api-endpoint", type=str)
+@click.option("--collection-name", type=str, default="collection_test")
+@click.option("--embedding-dimension", type=int, default=384)
+def run_check(token, api_endpoint, collection_name, embedding_dimension):
+    print(f"Checking contents of Astra DB collection: {collection_name}")
+
+    # Initialize our vector db
+    astra_db = AstraDB(token=token, api_endpoint=api_endpoint)
+    astra_db_collection = astra_db.collection(collection_name)
+
+    # Tally up the embeddings
+    docs_count = astra_db_collection.count_documents()
+    number_of_embeddings = docs_count["status"]["count"]
+
+    # Print the results
+    expected_embeddings = 3
+    print(
+        f"# of embeddings in collection vs expected: {number_of_embeddings}/{expected_embeddings}"
+    )
+
+    # Check that the assertion is true
+    assert number_of_embeddings == expected_embeddings, (
+        f"Number of rows in generated table ({number_of_embeddings})"
+        f"doesn't match expected value: {expected_embeddings}"
+    )
+
+    # Generate a random embedding of the appropriate length
+    random_vector = [round(random.uniform(0, 1), 1) for _ in range(embedding_dimension)]
+
+    find_one = astra_db_collection.find_one()
+    random_vector = find_one["data"]["document"]["vector"]
+    random_text = find_one["data"]["document"]["content"]
+    print(random_text)
+
+    # Perform a similarity search
+    find_result = astra_db_collection.vector_find(random_vector, limit=1)
+    print(find_result)
```

Comment:
@erichare Do you know why this is not finding any results? I'm essentially searching with the same vector I just pulled from the database. It should just pull up the same record. But the result is always blank.

Reply:
@potter-potter So, i'm looking at the test.... internally, the vector / embeddings column in astra is always named `$vector`, i.e., i would expect line 39 to look like.

```python
random_vector = find_one["data"]["document"]["$vector"]
```

The `find` and `vector_find` methods at the API level are going to look for that field. I'm wondering if we need to make a small change to the integration as a result?

Reply:
My guess would have been the `normalize_dict()` method but you'd probably know better? At the time i thought that was normalization for purposes of unstructured, but if thats what is being sent to Astra, with `write_dict`, i think that needs to be changed... that's my guess anyway. `vector` -> `$vector`

Reply:
Left a couple comments that I *think* will fix it... this was my oversight, if i'm correct about it.... thinking that the normalization of the dictionary was for the return to Unstructured rather than for insertion into Astra.

I think once these changes are made, `vector_find` should work properly.... as of now i think its failing because its not actually seeing the "proper" `$vector` field in the documents.

Reply:
Does the free account not allow vector search via api?

Reply:
It should allow you to! If you login to `astra.datastax.com` and click your Database, and the Data Explorer tab shows a collection with N dimensions (plus >= 1 records) then `vector_find` should work just fine!